drivers/ethernet/eth_mcux.c

```diff
@@ -601,9 +601,16 @@ static void eth_0_iface_init(struct net_if *iface)
 	context->iface = iface;
 }
 
-static struct net_if_api api_funcs_0 = {
-	.init	= eth_0_iface_init,
-	.send	= eth_tx,
+static enum eth_hw_caps eth_mcux_get_capabilities(struct device *dev)
+{
+	return 0;
+}
+
+static struct ethernet_api api_funcs_0 = {
+	.iface_api.init = eth_0_iface_init,
+	.iface_api.send = eth_tx,
```

Comment:
It's often better to initialize this using more compound statements, as uninitialized elements will be set to their zero value:
```C
.iface_api = {
    .init = eth_0_iface_init,
    .send = eth_tx
```

(It might not have anything now, but if the API is expanded in the future, makes things easier to check for in `eth.h`)